tests/testsuite/messages.rs

```diff
@@ -47,6 +45,13 @@ pub fn raw_rustc_output(project: &Project, path: &str, extra: &[&str]) -> String
     result
 }
 
+fn redact_rustc_message(msg: &str) -> String {
+    msg.replace("help:", "[HELP]")
+        .replace("note:", "[NOTE]")
+        .replace("warning:", "[WARNING]")
+        .replace("error:", "[ERROR]")
+}
```

Comment:
I'm not sure if removing the `raw_rustc_output` function and relying solely on snapshots is a viable option. Therefore, I've currently redacted the messages with it. However, even if removing `raw_rustc_output` is acceptable, there's another file, `cache_messages`, that depends on `raw_rustc_output`. Therefore, I believe it would be better to refactor it in a follow-up pull request.